### ISSUE TITLE
exclude_paths not working

### ISSUE THREAD

**Issue by djdanielsson** [ansible/ansible-lint#2687]
##### Summary

I am trying to exclude files from being checked and the linter still checks them.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.13.4]
  config file = /Users/ddaniels/workspace/forks/aap_configuration_template/ansible.cfg
  configured module search path = ['/Users/ddaniels/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ddaniels/workspace/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/ddaniels/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/ddaniels/workspace/ansible/bin/ansible
  python version = 3.10.8 (main, Oct 13 2022, 09:48:40) [Clang 14.0.0 (clang-1400.0.29.102)]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.8.6 using ansible 2.13.4

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

MacOS 13.0.1 and GitHub actions

##### STEPS TO REPRODUCE

You should be able to try adding exclude_paths to your ansible-lint.yml file and see if it skips them or not.

ansible-lint.yml file
```yaml
---
# Ansible Linter rules
# Mock modules or roles in order to pass ansible-playbook --syntax-check
mock_modules:
  - ansible.controller.controller_api

# Exclude paths from linter
exclude_paths:
  - '.github/'
  - 'changelogs/'
  - 'playbooks/install_configure.yml'

# Make output parsable
parseable: true

# Set output to quiet
quiet: false

# Path to rules dir
# rulesdir:

# Ansible Linter rules
skip_list:
  - yaml[colons] # Violations reported by yamllint.
  - yaml[line-length]  # Violations reported by yamllint.
  - yaml[new-line-at-end-of-file]  # Violations reported by yamllint.

warn_list:  # or 'skip_list' to silence them completely
  - git-latest # Allow for newest git version
  - package-latest # Allow newest package version
  - risky-file-permissions  # File permissions unset or incorrect.
  - risky-shell-pipe  # Shells that use pipes should set the pipefail option.
  - template-instead-of-copy  # Templated files should use template instead of copy
  - command-instead-of-shell  # Use shell only when shell functionality is required.

# Set verbosity
verbosity: 1
...
```

##### Desired Behavior

Skip the folders/files that are listed in exclude_paths.

##### Actual Behavior

Still checks folders/files that are listed to be excluded.

```console
$ ansible-lint --offline --profile=production -vvvv
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/Users/ddaniels/.cache/ansible-compat/129486', colored=True, configured=True, cwd=PosixPath('/Users/ddaniels/workspace/forks/aap_configuration_template'), display_relative_path=True, exclude_paths=['/Users/ddaniels/workspace/forks/aap_configuration_template/.config/.github', '/Users/ddaniels/workspace/forks/aap_configuration_template/.config/changelogs', '/Users/ddaniels/workspace/forks/aap_configuration_template/.config/playbooks/install_configure.yml'], format='rich', lintables=[], listrules=False, listtags=False, write_list=[], parseable=True, quiet=False, rulesdirs=['/Users/ddaniels/workspace/ansible/lib/python3.10/site-packages/ansiblelint/rules'], skip_list=['yaml[colons]', 'yaml[line-length]', 'yaml[new-line-at-end-of-file]'], tags=[], verbosity=5, warn_list=['latest[git]', 'package-latest', 'risky-file-permissions', 'risky-shell-pipe', 'template-instead-of-copy', 'command-instead-of-shell'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.{yaml,yml}'}, {'requirements': '**/meta/requirements.{yaml,yml}'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_filters=[], mock_modules=['ansible.controller.controller_api'], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=True, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, strict=False, rules={}, profile=['production'], progressive=False, rulesdir=[], use_default_rules=False, config_file='/Users/ddaniels/workspace/forks/aap_configuration_template/.config/ansible-lint.yml', version=False, cache_dir_lock=None)
DEBUG    /Users/ddaniels/workspace/forks/aap_configuration_template
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
DEBUG    Activating rule `avoid-dot-notation` due to profile `production`
DEBUG    Activating rule `disallowed-ignore` due to profile `production`
DEBUG    Activating rule `fqcn` due to profile `production`
DEBUG    Activating rule `import-task-no-when` due to profile `production`
DEBUG    Activating rule `meta-no-dependencies` due to profile `production`
DEBUG    Activating rule `single-entry-point` due to profile `production`
DEBUG    Activating rule `use-loop` due to profile `production`
DEBUG    Activating rule `galaxy` due to profile `shared`
DEBUG    Activating rule `ignore-errors` due to profile `shared`
DEBUG    Activating rule `layout` due to profile `shared`
DEBUG    Activating rule `meta-incorrect` due to profile `shared`
DEBUG    Activating rule `meta-no-info` due to profile `shared`
DEBUG    Activating rule `meta-no-tags` due to profile `shared`
DEBUG    Activating rule `meta-video-links` due to profile `shared`
DEBUG    Activating rule `meta-version` due to profile `shared`
DEBUG    Activating rule `meta-unsupported-ansible` due to profile `shared`
DEBUG    Activating rule `no-changed-when` due to profile `shared`
DEBUG    Activating rule `no-changelog` due to profile `shared`
DEBUG    Activating rule `no-handler` due to profile `shared`
DEBUG    Activating rule `no-relative-paths` due to profile `shared`
DEBUG    Activating rule `max-block-depth` due to profile `shared`
DEBUG    Activating rule `max-tasks` due to profile `shared`
DEBUG    Activating rule `unsafe-loop` due to profile `shared`
DEBUG    Activating rule `avoid-implicit` due to profile `safety`
DEBUG    Activating rule `latest` due to profile `safety`
DEBUG    Activating rule `package-latest` due to profile `safety`
DEBUG    Activating rule `risky-file-permissions` due to profile `safety`
DEBUG    Activating rule `risky-octal` due to profile `safety`
DEBUG    Activating rule `risky-shell-pipe` due to profile `safety`
DEBUG    Activating rule `name[template]` due to profile `moderate`
DEBUG    Activating rule `name[imperative]` due to profile `moderate`
DEBUG    Activating rule `name[casing]` due to profile `moderate`
DEBUG    Activating rule `no-free-form` due to profile `moderate`
DEBUG    Activating rule `spell-var-name` due to profile `moderate`
DEBUG    Activating rule `command-instead-of-module` due to profile `basic`
DEBUG    Activating rule `command-instead-of-shell` due to profile `basic`
DEBUG    Activating rule `deprecated-bare-vars` due to profile `basic`
DEBUG    Activating rule `deprecated-command-syntax` due to profile `basic`
DEBUG    Activating rule `deprecated-local-action` due to profile `basic`
DEBUG    Activating rule `deprecated-module` due to profile `basic`
DEBUG    Activating rule `inline-env-var` due to profile `basic`
DEBUG    Activating rule `key-order` due to profile `basic`
DEBUG    Activating rule `literal-compare` due to profile `basic`
DEBUG    Activating rule `jinja` due to profile `basic`
DEBUG    Activating rule `no-jinja-when` due to profile `basic`
DEBUG    Activating rule `no-tabs` due to profile `basic`
DEBUG    Activating rule `partial-become` due to profile `basic`
DEBUG    Activating rule `playbook-extension` due to profile `basic`
DEBUG    Activating rule `role-name` due to profile `basic`
DEBUG    Activating rule `schema` due to profile `basic`
DEBUG    Activating rule `name` due to profile `basic`
DEBUG    Activating rule `var-naming` due to profile `basic`
DEBUG    Activating rule `yaml` due to profile `basic`
DEBUG    Activating rule `internal-error` due to profile `min`
DEBUG    Activating rule `load-failure` due to profile `min`
DEBUG    Activating rule `parser-error` due to profile `min`
DEBUG    Activating rule `syntax-check` due to profile `min`
DEBUG    Unloading warning rule due to not being part of production profile.
DEBUG    Removing tag `experimental` from `avoid-implicit` rule because `production` profile makes it mandatory.
DEBUG    Unloading empty-string-compare rule due to not being part of production profile.
DEBUG    Removing tag `opt-in` from `galaxy` rule because `production` profile makes it mandatory.
DEBUG    Removing tag `experimental` from `galaxy` rule because `production` profile makes it mandatory.
DEBUG    Removing tag `experimental` from `ignore-errors` rule because `production` profile makes it mandatory.
DEBUG    Removing tag `experimental` from `key-order` rule because `production` profile makes it mandatory.
DEBUG    Unloading loop-var-prefix rule due to not being part of production profile.
DEBUG    Removing tag `experimental` from `no-free-form` rule because `production` profile makes it mandatory.
DEBUG    Unloading no-log-password rule due to not being part of production profile.
DEBUG    Unloading no-prompting rule due to not being part of production profile.
DEBUG    Unloading no-same-owner rule due to not being part of production profile.
DEBUG    Unloading only-builtins rule due to not being part of production profile.
DEBUG    Removing tag `experimental` from `risky-file-permissions` rule because `production` profile makes it mandatory.
DEBUG    Unloading run-once rule due to not being part of production profile.
DEBUG    Removing tag `experimental` from `schema` rule because `production` profile makes it mandatory.
DEBUG    Removing tag `experimental` from `var-naming` rule because `production` profile makes it mandatory.
DEBUG    40/48 rules included in the profile
WARNING  Collection install skipped because ansible versions before 2.14 do not support an offline mode.
INFO     Set ANSIBLE_LIBRARY=/Users/ddaniels/.cache/ansible-compat/129486/modules:/Users/ddaniels/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/ddaniels/.cache/ansible-compat/129486/collections:/Users/ddaniels/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/ddaniels/.cache/ansible-compat/129486/roles:/Users/ddaniels/workspace/forks/aap_configuration_template/roles
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
INFO     Executing syntax check on playbooks/custom_collection.yml (0.46s)
INFO     Executing syntax check on playbooks/install_aap.yml (0.46s)
INFO     Executing syntax check on playbooks/hub_config.yml (0.47s)
INFO     Executing syntax check on playbooks/build_ee.yml (0.51s)
INFO     Executing syntax check on playbooks/console_token_refresh.yml (0.51s)
INFO     Executing syntax check on playbooks/controller_config.yml (0.52s)
INFO     Executing syntax check on playbooks/install_configure.yml (0.52s)
WARNING  Listing 1 violation(s) that are fatal
playbooks/install_configure.yml:2:3: syntax-check[specific]: The field 'hosts' has an invalid value, which includes an undefined variable. The error was: 'env' is undefined
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - syntax-check[specific]  # Ansible syntax check failed.

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-command-syntax': (7, 'basic'), 'deprecated-local-action': (8, 'basic'), 'deprecated-module': (9, 'basic'), 'inline-env-var': (10, 'basic'), 'key-order': (11, 'basic'), 'literal-compare': (12, 'basic'), 'jinja': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name[template]': (23, 'moderate'), 'name[imperative]': (24, 'moderate'), 'name[casing]': (25, 'moderate'), 'no-free-form': (26, 'moderate'), 'spell-var-name': (27, 'moderate'), 'avoid-implicit': (28, 'safety'), 'latest': (29, 'safety'), 'package-latest': (30, 'safety'), 'risky-file-permissions': (31, 'safety'), 'risky-octal': (32, 'safety'), 'risky-shell-pipe': (33, 'safety'), 'galaxy': (34, 'shared'), 'ignore-errors': (35, 'shared'), 'layout': (36, 'shared'), 'meta-incorrect': (37, 'shared'), 'meta-no-info': (38, 'shared'), 'meta-no-tags': (39, 'shared'), 'meta-video-links': (40, 'shared'), 'meta-version': (41, 'shared'), 'meta-unsupported-ansible': (42, 'shared'), 'no-changed-when': (43, 'shared'), 'no-changelog': (44, 'shared'), 'no-handler': (45, 'shared'), 'no-relative-paths': (46, 'shared'), 'max-block-depth': (47, 'shared'), 'max-tasks': (48, 'shared'), 'unsafe-loop': (49, 'shared'), 'avoid-dot-notation': (50, 'production'), 'disallowed-ignore': (51, 'production'), 'fqcn': (52, 'production'), 'import-task-no-when': (53, 'production'), 'meta-no-dependencies': (54, 'production'), 'single-entry-point': (55, 'production'), 'use-loop': (56, 'production')}
                  Rule Violation Summary                   
 count tag                    profile rule associated tags 
     1 syntax-check[specific] min     core                 

Failed after : 1 failure(s), 0 warning(s) on 136 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by djdanielsson** [comment]
so in slack it was suggested to try '${PWD}/playbooks/install_configure.yml' which does appear to work. I find this weird because it clearly shows in the output that the path it finds the file with an issue is playbooks/install_configure.yml. I think either that output needs to be updated to the "full path" or exclude_paths should allow relative paths such as playbooks/install_configure.yml. Also the documentation does not appear to show this fix as well because it gives relative paths as well.

**Comment by branic** [comment]
The suggestion to add `${PWD}` came from issue #371. The [documentation](https://ansible-lint.readthedocs.io/configuring/) states that file locations are parsed using relative paths to the `.ansible-lint` file.

https://github.com/ansible/ansible-lint/blob/04763f182c0abc4ce35fe05f39b4c9759757f76f/.ansible-lint#L3-L10

I believe the correct resolution for this issue is to fix the relative path handling for the `exclude_paths` option so that it functions as documented.

**Comment by jmighion** [comment]
Seeing something similar, even with absolute paths: 
```
(3.10.0_dev) jmighion-work-mbp:ansible-azure-mgmt jmighion$ ansible-lint -p playbooks/configure_jumpbox.yml --exclude /Users/jmighion/.ansible/roles/ --exclude /Users/jmighion/Documents/git/ansible-azure-mgmt/roles/requirements.yml --exclude ${PWD}/roles/requirements.yml --offline
WARNING: PATH altered to include /Users/jmighion/.pyenv/versions/3.10.0/envs/3.10.0_dev/bin
WARNING  Ignored exception from RoleRelativePath.<bound method AnsibleLintRule.matchtasks of no-relative-paths: Doesn't need a relative path in role.>: argument of type 'NoneType' is not iterable
WARNING  Listing 768 violation(s) that are fatal
../../../.ansible/roles/exploide.dnf-automatic/defaults/main.yml:8: yaml: truthy value should be one of  (yaml[truthy])
../../../.ansible/roles/exploide.dnf-automatic/defaults/main.yml:9: yaml: truthy value should be one of  (yaml[truthy])
```

I have all the extra attempts to exclude since they aren't working, in hopes the overkill route would catch one.

**Comment by shatakshiiii** [comment]
Hey @djdanielsson!  Looks like the issue is being resolved with the latest version of ansible-lint. We could also see from the warning message shown in your logs that: `ansible versions before 2.14 do not support an offline mode`. 
So `exclude_paths` seems to ignore the mentioned `- 'playbooks/install_configure.yml'` with the updated version `6.10.0 a1`.

**Comment by evanpurkhiser** [comment]
FYI I am still seeing this issue

```
$ ansible-lint --version
ansible-lint 6.12.1 using ansible 2.14.2
```

```
ansible-lint
WARNING  Ignore loading rule from /opt/homebrew/Cellar/ansible-lint/6.12.1/libexec/lib/python3.11/site-packages/ansiblelint/rules/galaxy.py due to No module named 'pytest'
WARNING  Listing 2 violation(s) that are fatal
load-failure: Failed to load YAML file
roles/machines/server/files/appdaemon.yaml:1 could not determine a constructor for the tag '!secret'
  in "<unicode string>", line 11, column 14

load-failure: Failed to load YAML file
roles/machines/server/files/home-assistant.yaml:1 could not determine a constructor for the tag '!include_dir_merge_named'
  in "<unicode string>", line 23, column 11


             Rule Violation Summary
 count tag          profile rule associated tags
     2 load-failure min     core, unskippable

Failed after : 2 failure(s), 0 warning(s) on 73 files.
```

The following configuration **works**

```yml
exclude_paths:
  - "${PWD}/roles/machines/server/files/appdaemon.yaml"
  - "${PWD}/roles/machines/server/files/home-assistant.yaml"
```

and the following does not

```yml
exclude_paths:
  - roles/machines/server/files/appdaemon.yaml
  - roles/machines/server/files/home-assistant.yaml
```

**Comment by shaaga** [comment]
Its not working for me even with $PWD prepended in the paths

**Comment by djdanielsson** [comment]
You do not need it anymore
